crates/boundless-market/src/contracts/boundless_market.rs

```diff
@@ -209,6 +218,18 @@ fn extract_tx_log<E: SolEvent + Debug + Clone>(
     }
 }
 
+fn validate_fulfill_receipt(receipt: TransactionReceipt) -> Result<(), MarketError> {
+    if let Some(log) = receipt.decoded_log::<IBoundlessMarket::PaymentRequirementsFailed>() {
+        let raw_error = Bytes::copy_from_slice(&log.error);
+        match IBoundlessMarketErrors::abi_decode(&raw_error) {
+            Ok(err) => Err(MarketError::PaymentRequirementsFailed(err)),
+            Err(_) => Err(MarketError::PaymentRequirementsFailedUnknownError(raw_error)),
+        }
```

Comment:
Second guessing this change. Is perhaps a bit misleading to throw an error if only one of the fulfillments failed, especially since the tx didn't fail. Perhaps the sanity checks to avoid submitting if the order is fulfilled is sufficient? @mintybasil was there a strong reason why you wanted this, or just for a better error message? Wondering if okay to just warn log in these cases?